### PR TITLE
Modifications to work with pytorch 0.4

### PR DESCRIPTION
**Pull request by shawntan**
- Removed some `Variable`s
- Modified `embedding_dropout` so using `Functional`
- Modified `finetune.py` loss
- Modified `repackage_hidden`

### CODE REVIEW

**Comment by esvhd** [comment]
When I tried to run the original code with pytorch 0.4, in addition to the above, I also saw a warning that `torch.nn.functional.log_softmax()` in `splitcross.py` should now explicitly specify `dim=`. Any chance we can add this to this PR too? 
Thanks.

**Comment by esvhd** [comment]
Also should add `dim=-1` for all `torch.nn.functional.softmax()`, e.g. in `pointer.py`.

**Comment by keskarnitish** [comment]
@shawntan Thank you very much for this. https://github.com/salesforce/awd-lstm-lm/commit/457a422eb46e970a6aad659ca815a04b3d074d6c makes the repository PyTorch 0.4 compatible.
